frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx

```diff
@@ -98,6 +99,18 @@ export default class Dashboard extends Component {
     this.parametersAndCardsContainerRef = React.createRef();
   }
 
+  static getDerivedStateFromProps(props, state) {
+    const parametersList = getValuePopulatedParameters(
+      props.parameters,
+      props.parameterValues,
+    );
+    if (!_.isEqual(parametersList, state.parametersList)) {
+      return { parametersList };
+    }
+
+    return null;
+  }
+
```

Comment:
Seems like we only need a parameters count for the `stickyParameters` thing, so should we maybe just pass `props.parameters.length`? There won't be any need to do `getValuePopulatedParameters` too

Reply:
I'm on the fence, while I think passing the length is a nice optimization, the current code is already explicit and rather be on the "safe side".

Reply:
We might want to keep it that way because:
- `parametersList` is now used in [render()](https://github.com/metabase/metabase/pull/21848/files#diff-b27f144c8d7c51c6c4556af35ba48cb7b62b040e84b5a3d52ff9ac344c579310R237)
- we need the actual length of the `parametersList` generated from `getValuePopulatedParameters`